src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/specification/ReferralSpecifications.kt

```diff
@@ -0,0 +1,116 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.specification
+
+import org.springframework.data.jpa.domain.Specification
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AuthGroupID
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AuthUser
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.DynamicFrameworkContract
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.EndOfServiceReport
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Intervention
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Referral
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.ReferralAssignment
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.ServiceProvider
+import java.time.OffsetDateTime
+import java.util.UUID
+import javax.persistence.criteria.JoinType
+
+class ReferralSpecifications {
+  companion object {
+
+    fun sent(): Specification<Referral> {
+      return Specification<Referral> { root, query, cb ->
+        cb.isNotNull(root.get<OffsetDateTime>("sentAt"))
+      }
+    }
+
+    fun completed(): Specification<Referral> {
```

Comment:
The users would like to see prematurely_ended and completed referrals under the Completed dashboard. 

Do you think that delivered() would be a better terminology for this SQL?